context.go

```diff
@@ -85,6 +87,19 @@ func GetHeight(ctx Context) (int64, bool) {
 	return val, ok
 }
 
+// WithBlockTime sets the block time for the context. Block time is always
+// represented in UTC.
+func WithBlockTime(ctx Context, t time.Time) Context {
+	return context.WithValue(ctx, contextKeyTime, t.UTC())
+}
+
+// BlockTime returns current block wall clock time as declared in the context.
+// If time is not present in the context a zero time value is returned.
```

Comment:
Hmmm.... what about returning a second result (error or bool) to signify time is missing.
I would rather not accidentally use the zero time thinking it was set

Reply:
This is a good point. I followed the time code a bit and it is taken from the votes.precommits or genesis.
 https://github.com/tendermint/tendermint/blob/v0.29.1/types/block.go#L361
I am not sure if it is technical possible but it would be very unlikely that no validator sets any timestamp on his vote. Default is now: https://github.com/tendermint/tendermint/blob/v0.29.1/types/proposal.go#L42
But also note:
https://github.com/tendermint/tendermint/blob/v0.29.1/types/proposal.go#L68

Reply:
I can add a second return value as proposed. I also think that being explicit is good.

Reply:
My comment was about: we can safely assume time is always non zero (with default tendermint installations). No need for the additional bool

Reply:
Well, especially in test code, if this time is not explicitly set, you will blindly run against the zero value. That's what I wanted to avoid